src/main/java/core/basesyntax/FileWork.java

```diff
@@ -1,7 +1,51 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+
 public class FileWork {
+    private static final String SPECIFIED_CHARACTER = "w";
+    private static final String SPECIAL_CHARACTERS = "[,.\\s\\-:?!]";
+
     public String[] readFromFile(String fileName) {
-        return null;
+        StringBuilder filteredWords = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            String value = bufferedReader.readLine();
+            while (value != null) {
+                filteredWords.append(value.toLowerCase()).append(System.lineSeparator());
+                value = bufferedReader.readLine();
```

Comment:
```suggestion
            String line = bufferedReader.readLine();
            while (line != null) {
               String[] words = line.toLowerCase().split("\\W+");
                for (String word : words) {
                    if (word.startsWith(SPECIFIED_CHARACTER)) {
                        targetWords.append(word).append(" ");
                    }
                line = bufferedReader.readLine();
```
this way you won't need two loops below, you can split strings by space and sort them